python/lsst/pipe/tasks/characterizeImage.py

```diff
@@ -0,0 +1,493 @@
+            measuring and applying aperture correction is disabled because we don't have a final PSF
+        - if config.doMeasurePsf:
+            - measure PSF
+
+        @param[in,out] exposure  exposure to characterize (an lsst.afw.image.ExposureF or similar)
+            The following changes are made:
+            - update or set psf
+            - update detection and cosmic ray mask planes
+            - subtract background
+        @param[in] exposureIdInfo  ID info for exposure (an lsst.daf.butlerUtils.ExposureIdInfo)
+        @param[in,out] background  model of background model already subtracted from exposure
+            (an lsst.afw.math.BackgroundList).
+
+        @return pipe_base Struct containing these fields, all from the final iteration
+        of detect sources, measure sources and estimate PSF:
+        - exposure  characterized exposure; image is repaired by interpolating over cosmic rays,
+            mask is updated accordingly, and the PSF model is set
+        - sourceCat  detected sources (an lsst.afw.table.SourceCatalog)
+        - background  model of background subtracted from exposure (an lsst.afw.math.BackgroundList)
+        - psfCellSet  spatial cells of PSF candidates (an lsst.afw.math.SpatialCellSet)
+        """
+        # install a simple PSF model, if wanted
+        if self.config.doMeasurePsf:
+            if self.config.useSimplePsf or not exposure.hasPsf():
+                self.installSimplePsf.run(exposure=exposure)
+        elif not exposure.hasPsf():
+            raise pipeBase.TaskError("exposure has no PSF model and config.doMeasurePsf false")
+
+        # run repair, but do not interpolate over cosmic rays (do that elsewhere, with the final PSF model)
+        self.repair.run(exposure=exposure, keepCRs=True)
+        self.display("repair_iter", exposure=exposure)
+
+        damRes = self.detectAndMeasure.run(
+            exposure = exposure,
+            exposureIdInfo = exposureIdInfo,
+            background = background,
+            allowApCorr = False,  # wait for final PSF
+        )
+
+        measPsfRes = pipeBase.Struct(
+            cellSet = None
+        )
+        if self.config.doMeasurePsf:
+            if self.measurePsf.usesMatches:
+                matches = self.astrometry.loadAndMatch(
+                    exposure = exposure,
+                    sourceCat = damRes.sourceCat,
+                ).matches
+            else:
+                matches = None
+            measPsfRes = self.measurePsf.run(
+                exposure = exposure,
+                sources = damRes.sourceCat,
+                matches = matches,
+            )
+        self.display("measure_iter", exposure=exposure, sourceCat=damRes.sourceCat)
+
+        return pipeBase.Struct(
+            exposure = exposure,
+            sourceCat = damRes.sourceCat,
+            background = damRes.background,
+            psfCellSet = measPsfRes.cellSet,
```

Comment:
Why do we return the `psfCellSet`?  Is it ever going to be used?  If so, what might it be used for?
Are there more useful outputs from `measurePsf` that should be included?